packages/core/src/core/nonInteractiveToolExecutor.ts

```diff
@@ -4,166 +4,27 @@
-    let metadata: { [key: string]: any } = {};
-    if (
-      toolResult.error === undefined &&
-      typeof tool_display === 'object' &&
-      tool_display !== null &&
-      'diffStat' in tool_display
-    ) {
-      const diffStat = (tool_display as FileDiff).diffStat;
-      if (diffStat) {
-        metadata = {
-          ai_added_lines: diffStat.ai_added_lines,
-          ai_removed_lines: diffStat.ai_removed_lines,
-          user_added_lines: diffStat.user_added_lines,
-          user_removed_lines: diffStat.user_removed_lines,
-        };
-      }
-    }
-    const durationMs = Date.now() - startTime;
-    logToolCall(config, {
-      'event.name': 'tool_call',
-      'event.timestamp': new Date().toISOString(),
-      function_name: toolCallRequest.name,
-      function_args: toolCallRequest.args,
-      duration_ms: durationMs,
-      success: toolResult.error === undefined,
-      error:
-        toolResult.error === undefined ? undefined : toolResult.error.message,
-      error_type:
-        toolResult.error === undefined ? undefined : toolResult.error.type,
-      prompt_id: toolCallRequest.prompt_id,
-      metadata,
-      decision: ToolCallDecision.AUTO_ACCEPT,
-      tool_type:
-        typeof tool !== 'undefined' && tool instanceof DiscoveredMCPTool
-          ? 'mcp'
-          : 'native',
-    });
-
-    const response = convertToFunctionResponse(
-      toolCallRequest.name,
-      toolCallRequest.callId,
-      tool_output,
-    );
-
-    return {
-      callId: toolCallRequest.callId,
-      responseParts: response,
-      resultDisplay: tool_display,
-      error:
-        toolResult.error === undefined
-          ? undefined
-          : new Error(toolResult.error.message),
-      errorType:
-        toolResult.error === undefined ? undefined : toolResult.error.type,
-    };
-  } catch (e) {
-    const error = e instanceof Error ? e : new Error(String(e));
-    const durationMs = Date.now() - startTime;
-    logToolCall(config, {
-      'event.name': 'tool_call',
-      'event.timestamp': new Date().toISOString(),
-      function_name: toolCallRequest.name,
-      function_args: toolCallRequest.args,
-      duration_ms: durationMs,
-      success: false,
-      error: error.message,
-      error_type: ToolErrorType.UNHANDLED_EXCEPTION,
-      prompt_id: toolCallRequest.prompt_id,
-      tool_type:
-        typeof tool !== 'undefined' && tool instanceof DiscoveredMCPTool
-          ? 'mcp'
-          : 'native',
-    });
-    return {
-      callId: toolCallRequest.callId,
-      responseParts: [
-        {
-          functionResponse: {
-            id: toolCallRequest.callId,
-            name: toolCallRequest.name,
-            response: { error: error.message },
-          },
-        },
-      ],
-      resultDisplay: error.message,
-      error,
-      errorType: ToolErrorType.UNHANDLED_EXCEPTION,
-    };
-  }
+  return new Promise<ToolCallResponseInfo>((resolve, reject) => {
```

Comment:
Wow, this is so much more streamlined. I didn't realize that we had made coreToolScheduler so flexible to support all of this out of the box. Are there specific things that we had in non-Interactive that we didn't have in interactive?

Reply:
fantastic change. i love to see code deleted. just goes to show how many more places we could have had issues.